zenguard/zenguard.py

```diff
@@ -135,49 +137,21 @@ def detect(self, detectors: list[Detector], prompt: str):
                 headers={"x-api-key": self._api_key},
                 timeout=20,
             )
-            response.raise_for_status()
+            if response.status_code != 200:
+                raise RuntimeError(
+                    f"Received an unexpected status code: {response.status_code}\nResponse content: {response.json()}"
+                )
+            return response.json()
         except httpx.RequestError as e:
             raise RuntimeError(
                 f"An error occurred while making the request: {str(e)}"
             ) from e
-        except httpx.HTTPStatusError as e:
-            raise RuntimeError(
-                f"Received an unexpected status code: {response.status_code}\nResponse content: {response.json()}"
-            ) from e
-
-        return response.json()
 
     def detect_async(self, detectors: list[Detector], prompt: str):
```

Comment:
@nurvdil now `detect_async` is not really async? we should probably just delete this then and its tests.

Reply:
Fine. I just decided to do this so that people who used the detect_async method would not break anything

Reply:
I just thought that after this pull request I would make the detect_async method really asynchronous using aiohttp and asyncio.gather)